gph/src/ripser.h

```diff
@@ -995,24 +995,19 @@ class ripser
             index_t u = dset.find(v1), v = dset.find(v2);
 
             if (u != v) {
-                // link must be always done and now because we retrieve
-                // birth vertex value and index, it must be computed before
-                // the if condition
                 birth_vertex = dset.link_roots_and_get_birth_vertex(u, v);
-                if (get_diameter(e) != 0) {
```

Comment:
Note to self: leave inline comment explaining why we had to diverge from https://github.com/Ripser/ripser here.